revert.go

```diff
@@ -0,0 +1,102 @@
+package git
+
+/*
+#include <git2.h>
+*/
+import "C"
+import (
+	"runtime"
+)
+
+// RevertOptions contains options for performing a revert
+type RevertOptions struct {
+	Version      uint
+	Mainline     uint
+	MergeOpts    MergeOptions
+	CheckoutOpts CheckoutOpts
+}
+
+func (opts *RevertOptions) toC() *C.git_revert_options {
+	return &C.git_revert_options{
+		version:       C.uint(opts.Version),
+		mainline:      C.uint(opts.Mainline),
+		merge_opts:    *opts.MergeOpts.toC(),
+		checkout_opts: *opts.CheckoutOpts.toC(),
+	}
+}
+
+func revertOptionsFromC(opts *C.git_revert_options) RevertOptions {
+	return RevertOptions{
+		Version:      uint(opts.version),
+		Mainline:     uint(opts.mainline),
+		MergeOpts:    mergeOptionsFromC(&opts.merge_opts),
+		CheckoutOpts: checkoutOptionsFromC(&opts.checkout_opts),
+	}
+}
+
+func freeRevertOptions(opts *C.git_revert_options) {
+	freeCheckoutOpts(&opts.checkout_opts)
+}
+
+// DefaultRevertOptions initialises a RevertOptions struct with default values
+func DefaultRevertOptions() (RevertOptions, error) {
+	opts := C.git_revert_options{}
+
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+
+	ecode := C.git_revert_init_options(&opts, C.GIT_REVERT_OPTIONS_VERSION)
+	if ecode < 0 {
+		return RevertOptions{}, MakeGitError(ecode)
+	}
+
+	defer freeRevertOptions(&opts)
+	return revertOptionsFromC(&opts), nil
+}
+
+// Revert the provided commit leaving the index updated with the results of the revert
+func (r *Repository) Revert(commit *Commit, revertOptions *RevertOptions) error {
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+
+	var cOpts *C.git_revert_options
+
+	if revertOptions != nil {
+		cOpts = revertOptions.toC()
+		defer freeRevertOptions(cOpts)
+	}
+
+	ecode := C.git_revert(r.ptr, commit.cast_ptr, cOpts)
+	runtime.KeepAlive(commit)
```

Comment:
oh missed something: you also need

```go
runtime.KeepAlive(r)
```

here and in `RevertCommit`.

Reply:
Thanks for the review, `runtime.KeepAlive(r)` has been added.